.github/workflows/ci.yml

```diff
@@ -83,15 +83,33 @@ jobs:
       flink-version: "v1_20"
       http-client: ${{ matrix.http-client }}
       test: test_application_operations.sh
+  java_rutimes:
+    name: Java runtimes smoke test
+    needs: e2e_smoke_test
+    strategy:
+      matrix:
+        http-client: [ "okhttp" ]
+        java-version: [ "11", "17"]
+        flink-version:
+          - "v1_20"
+          - "v1_19"
+          - "v1_18"
+    uses: ./.github/workflows/e2e.yaml
+    with:
+      java-version: ${{ matrix.java-version }}
```

Comment:
I'm wondering if it's a bit funky that we are using `java-version` to control both the JDK/JRE used to build/run the operator and the runtime of the flink image. Aren't they different dimensions of the matrix?

This would show that a JDK17 operator works with a JDK17 flink image, but not cover the current default operator runtime with a JDK17 flink image.

Reply:
Yeah, I had debated making a `flink-tag` parameter but didn't as we would then have to manually encode the matrix. 

>This would show that a JDK17 operator works with a JDK17 flink image, but not cover the current default operator runtime with a JDK17 flink image.

I'm not convinced that the operator JDK really matters so don't think its particularly important to have a JDK11 operator deploying a JDK 17 Flink. I can go that direction if others see it as valuable.

Reply:
yeah you're right we aren't likely to see some incompatibility arising from different JRE combinations. Maybe it would be better to hold the operator JDK constant and vary only the flink runtime. Just to make it clearer which thing is varied.

Reply:
Yeah I agree with @SamBarker. The issue here is actually not the interplay of JDK versions but rather the config supplied by the operator's logic not being compatible (or more accuratly _enabling_) Java 17 support. So that's what we need to test.